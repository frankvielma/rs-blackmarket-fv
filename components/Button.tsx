import Link from 'next/link'

export default function Button({ name = '', link = '' }) {
  return (
    <Link href={link} className="focus:outline-none">
      <div className="custom-focus custom-active custom-hover btn-hover mx-[34px] mt-4 h-[44px] w-auto rounded-md border border-dark-violet bg-white p-2 text-center font-bold text-black active:bg-hover">
        <button className="focus:outline-none">{name}</button>
      </div>
    </Link>
  )
}
