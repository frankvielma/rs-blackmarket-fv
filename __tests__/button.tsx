import { render } from '@testing-library/react'
import Button, { Variants, Sizes } from '@/components/Button'

describe('Button', () => {
  it('renders button with primary variant, disabled and bold label', () => {
    const { container } = render(
      <Button
        variant={Variants.Primary}
        size={Sizes.Default}
        disabled={true}
        bold={true}
      />
    )
    expect(container).toMatchSnapshot()
  }),
    it('renders button with primary variant, enabled and not bold label', () => {
      const { container } = render(
        <Button
          variant={Variants.Primary}
          size={Sizes.Medium}
          disabled={false}
          bold={false}
        />
      )
      expect(container).toMatchSnapshot()
    }),
    it('renders button with outline variant, disabled and not bold label', () => {
      const { container } = render(
        <Button
          variant={Variants.Primary}
          size={Sizes.Large}
          disabled={true}
          bold={false}
        />
      )
      expect(container).toMatchSnapshot()
    })
})
