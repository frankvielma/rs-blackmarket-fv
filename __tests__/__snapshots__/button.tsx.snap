// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button with outline variant, disabled and not bold label 1`] = `
<div>
  <button
    class="h-[44px] rounded-md border text-center text-base
    font-normal
    bg-white text-black border-dark-violet
    bg-light-gray text-dark-gray outline-none cursor-not-allowed
  "
    type="button"
  />
</div>
`;

exports[`Button renders button with primary variant, disabled and bold label 1`] = `
<div>
  <button
    class="h-[44px] rounded-md border text-center text-base
    font-bold
    bg-black text-white border-white
    bg-light-gray text-dark-gray outline-none cursor-not-allowed
  "
    type="button"
  />
</div>
`;

exports[`Button renders button with primary variant, enabled and not bold label 1`] = `
<div>
  <button
    class="h-[44px] rounded-md border text-center text-base
    font-normal
    bg-black text-white border-white
    active:ring-2 active:ring-offset-2 active:outline-none custom-focus hover:bg-hover active:bg-hover
  "
    type="button"
  />
</div>
`;
